pkg/registry/minion/rest.go

```diff
@@ -91,6 +91,9 @@ func NodeToSelectableFields(node *api.Node) labels.Set {
 
 // MatchNode returns a generic matcher for a given label and field selector.
 func MatchNode(label labels.Selector, field fields.Selector) generic.Matcher {
+	if name, ok := field.RequiresExactMatch("name"); ok {
```

Comment:
I think it should be name. This is related to how fields are translated into lables in the conversion code, e.g.:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/api/v1beta3/conversion.go#L25
For all: ReplicationController, Pod and Node we have "name". 
But I agree with you that translating it to "metadata.name" would probably be better.